src/PublicSuffixListManager.php

```diff
@@ -57,101 +46,94 @@ class PublicSuffixListManager
+     * @param CacheInterface $cacheAdapter
+     * @param HttpAdapter    $httpAdapter
+     * @param string         $sourceUrl
      */
-    public function __construct(HttpAdapter $httpAdapter, string $cacheDir = null)
-    {
-        $this->cacheDir = $cacheDir ?? realpath(dirname(__DIR__) . DIRECTORY_SEPARATOR . 'data');
+    public function __construct(
+        CacheInterface $cacheAdapter,
+        HttpAdapter $httpAdapter,
+        string $sourceUrl = self::PUBLIC_SUFFIX_LIST_URL
+    ) {
+        $this->cacheAdapter = $cacheAdapter;
         $this->httpAdapter = $httpAdapter;
+        $this->sourceUrl = $sourceUrl;
     }
 
     /**
      * Gets Public Suffix List.
      *
-     * @param string $list the Public Suffix List type
+     * @param string $type the Public Suffix List type
      *
      * @return PublicSuffixList
      */
-    public function getList($list = self::ALL_DOMAINS): PublicSuffixList
+    public function getList($type = self::ALL_DOMAINS): PublicSuffixList
     {
-        $cacheBasename = isset(self::$domainList[$list]) ? self::$domainList[$list] : self::PDP_PSL_PHP_FILE;
-        $cacheFile = $this->cacheDir . '/' . $cacheBasename;
-        if (!file_exists($cacheFile)) {
+        static $type_lists = [
+            self::ALL_DOMAINS => self::ALL_DOMAINS,
+            self::ICANN_DOMAINS => self::ICANN_DOMAINS,
+            self::PRIVATE_DOMAINS => self::PRIVATE_DOMAINS,
+        ];
+
+        $type = $type_lists[$type] ?? self::ALL_DOMAINS;
+        $list = $this->cacheAdapter->get($type);
+        if ($list === null) {
             $this->refreshPublicSuffixList();
+            $list = $this->cacheAdapter->get($type);
         }
 
-        return new PublicSuffixList($cacheFile);
+        return new PublicSuffixList(json_decode($list, true));
     }
 
     /**
      * Downloads Public Suffix List and writes text cache and PHP cache. If these files
      * already exist, they will be overwritten.
-     */
-    public function refreshPublicSuffixList()
-    {
-        $publicSuffixList = $this->httpAdapter->getContent(self::PUBLIC_SUFFIX_LIST_URL);
-        $this->cache(self::PDP_PSL_TEXT_FILE, $publicSuffixList);
-
-        $publicSuffixListArray = $this->convertListToArray();
-        foreach ($publicSuffixListArray as $type => $list) {
-            $content = '<?php' . PHP_EOL . 'return ' . var_export($list, true) . ';';
-            $this->cache(self::$domainList[$type], $content);
-        }
-    }
-
-    /**
-     * Cache content to disk.
      *
-     * @param string $basename basename in cache dir where data will be written
-     * @param string $data     data to write
+     * Returns true if all list are correctly refreshed
      *
-     * @throws Exception if unable to write file
-     *
-     * @return int Number of bytes that were written to the file
+     * @return bool
      */
-    private function cache(string $basename, string $data): int
+    public function refreshPublicSuffixList(): bool
```

Comment:
Since refreshPublicSuffixList is always called if a there's a cache miss. There is no need to cache the raw public suffix list only the parsed json representation are cached.